### PR TITLE
Added BYOL model

### PR DESCRIPTION
**Pull request by pranavsinghps1**
Added Bootstrap Your Own Latent Model (https://arxiv.org/abs/2006.07733)

### CODE REVIEW

**Comment by IgorSusmelj** [comment]
Hey @pranavsinghps1,
thanks a lot for the PR. Will have a look at it today :) 

That's very good timing as I wanted to look into the Imagenette benchmark issue (https://github.com/lightly-ai/lightly/issues/279)

File: lightly/models/byol.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            Dimension of the hidden layer in the projection and prediction mlp.
+        out_dim:
+            Dimension of the output (after the projection/prediction mlp).
+        m:
+            Momentum for the momentum update of encoder.
+
+    """
+
+    def __init__(self,
+
```
Comment (philippmwirth) [inline lightly/models/byol.py:43]: Please remove this empty line

File: lightly/models/byol.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        
+            if return_features:
+                out1 = (out1, f1)
+        
+        return out0, out1
+
+    def forward(self,
+                x0: torch.Tensor,
+                x1: torch.Tensor = None,
+
```
Comment (philippmwirth) [inline lightly/models/byol.py:129]: Please add the `return_features: bool = False` flag again and pass it to `_forward` in the lines 148, 149 below.

File: lightly/models/byol.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        Performs two forward passes, once where x0 is passed through the encoder
+        and x1 through the momentum encoder and once the other way around.
+
+        Args:
+            x0:
+                Tensor of shape bsz x channels x W x H.
+            x1:
+                Tensor of shape bsz x channels x W x H.
+
```
Comment (philippmwirth) [inline lightly/models/byol.py:141]: Please add a docstring entry about the `return_features` flag. You can take the one from `_forward`

File: lightly/models/byol.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        Returns: 
+                tensors after passing through the online and
```
Comment (philippmwirth) [inline lightly/models/byol.py:143]: Start the sentence with a capital letter and it's good :)

**Review by philippmwirth — COMMENTED** [review]
Hi @pranavsinghps1, thanks a lot for your contribution :)

I have left some comments about minor issues with the code, please take a look at them.

Could you please also provide some basic tests? If you don't know where to start, check out our tests for MoCo:
https://github.com/lightly-ai/lightly/blob/911141997ec0fa93bab46357a156474efdc3676d/tests/models/test_ModelsMoCo.py#L23

Don't hesitate to reach out if anything's unclear, I'm here to help :)

File: lightly/models/byol.py (hunk 4 above)
Comment (pranavsinghps1) [reply lightly/models/byol.py]: Addressed all the above things with commit d40a850

**Comment by IgorSusmelj** [comment]
Close the PR because it's a duplicate of https://github.com/lightly-ai/lightly/pull/348

**Pull request closed without merging**